_staging/src/k8s.io/client-go/copy.sh

```diff
@@ -0,0 +1,153 @@
+CLIENT_REPO_FROM_SRC=${2:-"k8s.io/client-go"}
+CLIENT_REPO=${MAIN_REPO}/_staging/src/${CLIENT_REPO_FROM_SRC}
+CLIENT_REPO_TEMP=${CLIENT_REPO}/_tmp
+
+# working in the ${CLIENT_REPO_TEMP} so 'godep save' won't complain about dirty working tree.
+echo "creating the _tmp directory"
+mkdir -p ${CLIENT_REPO_TEMP}
+
+# mkcp copies file from the main repo to the client repo, it creates the directory if it doesn't exist in the client repo.
+function mkcp() {
+    mkdir -p "${CLIENT_REPO_TEMP}/$2" && cp -r "${MAIN_REPO}/$1" "${CLIENT_REPO_TEMP}/$2"
+}
+
+echo "copying client packages"
+mkcp "pkg/client/clientset_generated/release_1_4" "pkg/client/clientset_generated"
+mkcp "/pkg/client/record/" "/pkg/client"
+mkcp "/pkg/client/cache/" "/pkg/client"
+# TODO: make this test file not depending on pkg/client/unversioned
+rm ${CLIENT_REPO_TEMP}/pkg/client/cache/listwatch_test.go
+mkcp "/pkg/client/restclient" "/pkg/client"
+mkcp "/pkg/client/testing" "/pkg/client"
+# remove this test because it imports the internal clientset
+rm ${CLIENT_REPO_TEMP}/pkg/client/testing/core/fake_test.go
+mkcp "/pkg/client/transport" "/pkg/client"
+mkcp "/pkg/client/typed" "/pkg/client"
+
+mkcp "/pkg/client/unversioned/auth" "/pkg/client/unversioned"
+mkcp "/pkg/client/unversioned/clientcmd" "/pkg/client/unversioned"
+mkcp "/pkg/client/unversioned/portforward" "/pkg/client/unversioned"
+# remove this test because it imports the internal clientset
+rm ${CLIENT_REPO_TEMP}/pkg/client/unversioned/portforward/portforward_test.go
+
+pushd ${CLIENT_REPO_TEMP}
+echo "generating vendor/"
+GO15VENDOREXPERIMENT=1 godep save ./...
+popd
+
+echo "move to the client repo"
+# clean the ${CLIENT_REPO}
+ls ${CLIENT_REPO} | grep -v 'copy.sh\|_tmp' | xargs rm -r
+mv ${CLIENT_REPO_TEMP}/* ${CLIENT_REPO}
+rm -r ${CLIENT_REPO_TEMP}
+rm -r ${CLIENT_REPO}/Godeps
+
+echo "moving vendor/k8s.io/kuberentes"
+cp -rn ${CLIENT_REPO}/vendor/k8s.io/kubernetes/. ${CLIENT_REPO}/
+# client-go will share the vendor of the main repo for now. When client-go
+# becomes a standalone repo, it will have its own vendor
+rm -r ${CLIENT_REPO}/vendor
+# remove the pkg/util/net/sets/README.md to silent hack/verify-munge-docs.sh
+# TODO: probably we should convert the README.md a doc.go
+find ./ -name "README.md" | xargs rm
+
+echo "rewriting imports"
+grep -Rl "\"${MAIN_REPO_FROM_SRC}" ./ | grep ".go" | grep -v "vendor/" | xargs sed -i "s|\"${MAIN_REPO_FROM_SRC}|\"${CLIENT_REPO_FROM_SRC}|g"
+
+echo "converting pkg/client/record to v1"
+# need a v1 version of ref.go
+cp ${CLIENT_REPO}/pkg/api/ref.go ${CLIENT_REPO}/pkg/api/v1/ref.go
+gofmt -w -r 'api.a -> v1.a' ${CLIENT_REPO}/pkg/api/v1/ref.go
+gofmt -w -r 'Scheme -> api.Scheme' ${CLIENT_REPO}/pkg/api/v1/ref.go
+# rewriting package name to v1
+sed -i 's/package api/package v1/g' ${CLIENT_REPO}/pkg/api/v1/ref.go
+# ref.go refers api.Scheme, so manually import /pkg/api
+sed -i "s,import (,import (\n\"${CLIENT_REPO_FROM_SRC}/pkg/api\",g" ${CLIENT_REPO}/pkg/api/v1/ref.go
```

Comment:
I was under the impression we should not rely on `goimports` in our tool. A quick grep shows `go-to-protobuf` uses goimports but that tool is containerized.